open_event/views/admin/track_view.py

```diff
@@ -0,0 +1,80 @@
+from flask_admin import  expose
+from flask_admin.contrib.sqla import ModelView
+from open_event.forms.admin.track_form import TrackForm
+from ...helpers.helpers import is_track_name_unique_in_event
+from ...helpers.data_getter import DataGetter
+from open_event.models.track import Track
+from flask import request, url_for, redirect, flash
+from ...helpers.data import DataManager
+from flask.ext import login
+
+
+class TrackView(ModelView):
+    @expose('/')
+    def index_view(self, event_id):
+        """Track list view"""
+        tracks = DataGetter.get_tracks(event_id)
+        events = DataGetter.get_all_events()
+        return self.render('admin/model/track/list1.html',
+                           objects=tracks,
+                           event_id=event_id,
+                           events=events)
+
+    @expose('/new', methods=('GET', 'POST'))
+    def create_view(self, event_id):
+        """New track view"""
+        events = DataGetter.get_all_events()
+        form = TrackForm(request.form)
+        if form.validate_on_submit():
+            if is_event_admin_or_editor(event_id) and is_track_name_unique_in_event(form, event_id):
+                DataManager.create_track(form, event_id)
+                flash("Track added")
+            else:
+                flash("You don't have permission!")
+            return redirect(url_for('.index_view', event_id=event_id))
+        return self.render('admin/model/create_model.html',
+                           form=form,
+                           event_id=event_id,
+                           events=events,
+                           cancel_url=url_for('.index_view',
+                                              event_id=event_id))
+
+    @expose('/edit', methods=('GET', 'POST'))
+    def edit_view(self, event_id):
+        """Edit track view"""
+        track_id = request.args.get('track_id', '')
+        track = DataGetter.get_object(Track, track_id)
+        events = DataGetter.get_all_events()
+        form = TrackForm(obj=track)
+        if form.validate_on_submit() and is_track_name_unique_in_event(form, event_id, track_id):
+            if is_event_admin_or_editor(event_id):
+                DataManager.update_track(form, track)
+                flash("Track updated")
+            else:
+                flash("You don't have permission!")
+            return redirect(url_for('.index_view', event_id=event_id))
+        return self.render('admin/model/create_model.html',
+                           form=form,
+                           event_id=event_id,
+                           events=events,
+                           cancel_url=url_for('.index_view', event_id=event_id))
+
+    @expose('/delete', methods=('GET', 'POST'))
+    def delete_view(self, event_id):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2530633423/issues/source?bid=3233864&fileBranchId=3283317#l63)